src/LinuxTracing/SwitchesStatesNamesVisitorTest.cpp

```diff
@@ -185,13 +226,27 @@ ::testing::Matcher<SchedulingSlice> SchedulingSliceEq(const SchedulingSlice& exp
                           expected.out_timestamp_ns()));
 }
 
-ThreadStateSlice MakeThreadStateSlice(uint32_t tid, ThreadStateSlice::ThreadState thread_state,
-                                      uint64_t duration_ns, uint64_t end_timestamp_ns) {
+ThreadStateSlice MakeThreadStateSlice(
+    uint32_t tid, ThreadStateSlice::ThreadState thread_state, uint64_t duration_ns,
+    uint64_t end_timestamp_ns,
+    ThreadStateSlice::WakeupReason wakeup_reason = ThreadStateSlice::kNotApplicable,
+    uint32_t wakeup_tid = 0, uint32_t wakeup_pid = 0,
+    bool has_switch_out_or_wakeup_callstack = false) {
```

Comment:
I don't think this function should have this parameter. Rather, we should pass a `ThreadStateSlice::CallstackStatus` directly. We should probably also pass the callstack id. All the other fields are passed and copied as-is: this is just a utility method to construct a `ThreadStateSlice` in a single call rather than calling all the setter methods.

Reply:
Done. I kept the id as 0, though. I don't see it ever being set differently here in this context.